apps/dashboard/src/components/contract-components/contract-publish-form/extensions-param-selector.tsx

```diff
@@ -0,0 +1,48 @@
+import { Flex } from "@chakra-ui/react";
+import { Select } from "chakra-react-select";
+import type React from "react";
+import { useMemo } from "react";
+import type {
+  useConstructorParamsFromABI,
+  useFunctionParamsFromABI,
+} from "../hooks";
+
+interface ExtensionsParamSelectorProps {
+  deployParams:
+    | ReturnType<typeof useFunctionParamsFromABI>
+    | ReturnType<typeof useConstructorParamsFromABI>;
+  value: string;
+  onChange: (fn: string) => void;
+}
+
+export const ExtensionsParamSelector: React.FC<
+  ExtensionsParamSelectorProps
+> = ({ deployParams, value, onChange }) => {
+  const options = useMemo(() => {
+    return deployParams.map((f) => ({
+      label: f.name,
+      value: f.name,
+    }));
+  }, [deployParams]);
+
+  return (
+    <Flex gap={2} alignItems="center" w="full">
+      <Select
```

Comment:
if at all possible use shad + tw here already, let's try avoiding anything *new* that still uses chakra